core/src/services/webhdfs/writer.rs

```diff
@@ -64,6 +87,69 @@ impl oio::OneShotWrite for WebhdfsWriter {
             _ => Err(parse_error(resp).await?),
         }
     }
+
+    async fn complete_block(&self, block_ids: Vec<String>) -> Result<()> {
+        let Some(ref atomic_write_dir) = self.backend.atomic_write_dir else {
+            return Err(Error::new(
+                ErrorKind::Unsupported,
+                "write multi is not supported when atomic is not set",
+            ));
+        };
+        let path = block_ids[0].clone();
+        if block_ids.len() >= 2 {
+            let sources: Vec<String> = block_ids[1..]
+                .iter()
+                .map(|s| format!("{}{}", atomic_write_dir, s))
+                .collect();
+            // concat blocks
+            let req = self
+                .backend
+                .webhdfs_concat_request(&format!("{}{}", atomic_write_dir, path), sources)?;
```

Comment:
What will happen if users write `path` failed and start write it again? We should generate a path for this tmp file.

Actually, we can merge all the files into the first block and then rename it to the desired path.

Reply:
`Actually, we can merge all the files into the first block and then rename it to the desired path.`

I'm doing it just like this.